### PR TITLE
auto OCP-33618

### PR DESCRIPTION
**Pull request by huiran0826**
Auto OCP-33618

Logs: https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3-smoke/2473/console

Can you help take a look? Thanks!
@rbbratta @weliang1 @anuragthehatter @zhaozhanqi

### CODE REVIEW

File: features/networking/ovn-egress-ip.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Feature: OVN Egress IP related features
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-33618
+  @admin
+  @destructive
+  Scenario: EgressIP works for all pods in the matched namespace when only configure namespaceSelector
+    Given I save ipecho url to the clipboard
```
Comment (zhaozhanqi) [inline features/networking/ovn-egress-ip.feature:8]: so this mean this case only can be tested on vsphere cluster for now. 
if so, please also update case filter.

Comment (huiran0826) [reply features/networking/ovn-egress-ip.feature:8]: Yes, this auto case can only be run in vSphere.  Will update the case filter in polarion.  Thanks for reminder.

File: features/step_definitions/networking.rb [hunk 2]
```diff
@@ -628,7 +628,7 @@
       pod.node_name == node_name
     }.first
     cache_resources ovnkube_pod
-    @result = ovnkube_pod.exec(network_cmd, as: admin)
+    @result = ovnkube_pod.exec(network_cmd, container: "ovn-controller", as: admin)
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:631]: I forgot they added `kube-rbac-proxy` container to `ovnkube-node` pod as well.  So we need to specify the container. :+1:

File: features/step_definitions/networking.rb [hunk 3]
```diff
@@ -1253,3 +1253,18 @@
   end
   logger.info "The egressfirewall type is stored to the #{cb_name} clipboard."
 end
+
+Given /^I save ipecho url to the#{OPT_SYM} clipboard$/ do | cb_name |
+  ensure_admin_tagged
+  cb_name = "cb_ipecho_url" unless cb_name
+  cb[cb_name]="172.31.249.80:9095"
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:1260]: maybe we should add a comment about who owns this service, so we know who to contact?  Or is it documented somewhere else?

Comment (huiran0826) [reply features/step_definitions/networking.rb:1260]: @rbbratta Thanks Ross! The ipecho  is running on a VM in vmc network, and currently maintained by me.  :)

File: features/networking/ovn-egress-ip.feature [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      | name     | test-rc                |
+      | replicas | 3                      |
+    Then the step should succeed
+    Given 3 pods become ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(3).name` is stored in the :pod3 clipboard
+
+    # Check source ip is egress ip
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | curl | -s | --connect-timeout | 5 | <%= cb.cb_ipecho_url %> |
```
Comment (liangxia) [inline features/networking/ovn-egress-ip.feature:58]: Perhaps use name `ipecho_url`, so it's `<%= cb.ipecho_url %>`.

File: features/networking/ovn-egress-ip.feature [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    And I replace lines in "egressip1.yaml":
+      | 172.31.249.227 | "<%= cb.valid_ip %>" |
+    And I run the :create admin command with:
+      | f | egressip1.yaml |
+    Then the step should succeed
+    Given I register clean-up steps:
+    """
+    Given the ovn egressip objects are removed
+    Then the step should succeed
+    """
```
Comment (liangxia) [inline features/networking/ovn-egress-ip.feature:44]: Above 5 lines can be shorten as `And admin ensures "xxx" egress_ip is deleted after scenario`.

You can refer to `lib/openshift/egress_network_policy.rb` to write the `egress_ip.rb`

File: features/networking/ovn-egress-ip.feature (hunk 4 above)
Comment (huiran0826) [reply features/networking/ovn-egress-ip.feature]: Thanks, updated it.

File: features/networking/ovn-egress-ip.feature (hunk 5 above)
Comment (huiran0826) [reply features/networking/ovn-egress-ip.feature]: Good suggestion, fixed it.
 [10:37:38] INFO> Shell Commands: oc delete egressip egressip --wait=false --kubeconfig=/Users/hrwang/workdir/huiran-mac-hrwang/ocp4_admin.kubeconfig
      egressip.k8s.ovn.org "egressip" deleted
      
      [10:37:39] INFO> Exit Status: 0

File: features/step_definitions/networking.rb (hunk 3 above)
Comment (akostadinov) [reply features/step_definitions/networking.rb:1260]: @huiran0826 , this `ipecho` service needs some description of what it is and what is it used for. Doing a quick search I can't for sure know what this should mean here.

Comment (huiran0826) [reply features/step_definitions/networking.rb:1260]: @akostadinov This ipecho service is used for egress ip feature testing.  It echos the source IP of the outbound traffic from cluster.  I will add comments about this step in next PR.  Thanks!

**Pull request merged**